pretyping/typing.mli

```diff
@@ -49,6 +49,8 @@ val check_type_fixpoint : ?loc:Loc.t -> env -> evar_map ->
 (** Variant of {!check} that assumes that the argument term is well-typed. *)
 val check_actual_type : env -> evar_map -> unsafe_judgment -> types -> evar_map
 
+val type_judgment : env -> evar_map -> unsafe_judgment -> evar_map * unsafe_type_judgment
```

Comment:
This API goes against the unwritten invariant that inhabitants of `unsafe_judgment` are well-typed.

Reply:
WDYM?

Reply:
It should take `constr -> types -> ...` instead of `unsafe_judgment -> ...`. Despite the name, in almost all places this type is used to signal that some type-checking has been performed and that we know that the value has the type of the type.

Reply:
This API does assume that though

Reply:
Oh, right, I misread what this was doing. LGTM then.